pkg/client/dtclient/client.go

```diff
@@ -352,20 +347,18 @@ func NewPlatformClient(dtURL string, token string, oauthCredentials auth.OauthCr
 }
 
 // NewClassicClient creates a new dynatrace client to be used for classic environments
-func NewClassicClient(dtURL string, token string, opts ...func(dynatraceClient *DynatraceClient)) (*DynatraceClient, error) {
+func NewClassicClient(dtURL string, client *rest.Client, opts ...func(dynatraceClient *DynatraceClient)) (*DynatraceClient, error) {
 	dtURL = strings.TrimSuffix(dtURL, "/")
 	if err := validateURL(dtURL); err != nil {
 		return nil, err
 	}
 
-	tokenClient := auth.NewTokenAuthClient(token)
-	restClient := rest.NewRestClient(tokenClient, rest.CreateRateLimitStrategy())
 	d := &DynatraceClient{
 		serverVersion:         version.Version{},
 		environmentURL:        dtURL,
 		environmentURLClassic: dtURL,
-		client:                restClient,
-		clientClassic:         restClient,
+		client:                client,
```

Comment:
how does the DynatraceClient behave when `client` and `environmentURL` are not set?
Any platform related call will not work anyway (and hopefully not happen) for a classic environment, so can't we just leave these uninitialized rather than setting wrong data?

Reply:
I've changed it, it required some minor adoptions to the test code, let's see if it breaks something :D